### PR TITLE
Update to Scala 2.13.0-RC1

### PR DESCRIPTION
**Pull request by raboof**
Prepare the `release-2.5` branch for releasing with scala 2.13.0-RC1

Depends on https://github.com/scala/scala-java8-compat/pull/139

### CODE REVIEW

**Comment by lrytz** [comment]
https://oss.sonatype.org/content/repositories/orgscala-lang-1731 has scala-java8-compat 0.9.0 for 2.13.0-RC1. Let me know if it works, then I'll release it.

**Comment by raboof** [comment]
> https://oss.sonatype.org/content/repositories/orgscala-lang-1731 has scala-java8-compat 0.9.0 for 2.13.0-RC1. Let me know if it works, then I'll release it.

Looks good!

**Comment by lrytz** [comment]
> Looks good!

It's on maven central http://repo1.maven.org/maven2/org/scala-lang/modules/scala-java8-compat_2.13.0-RC1/0.9.0/

File: akka-distributed-data/src/test/scala/akka/cluster/ddata/ORSetSpec.scala [hunk 1]
```diff
@@ -599,7 +599,7 @@ class ORSetSpec extends WordSpec with Matchers {
     }
 
     "have unapply extractor" in {
-      val s1 = ORSet.empty.add(node1, "a").add(node2, "b")
+      val s1 = ORSet.empty[String].add(node1, "a").add(node2, "b")
```
Comment (patriknw) [inline akka-distributed-data/src/test/scala/akka/cluster/ddata/ORSetSpec.scala]: I'd say that it's a good thing to be explicit about the type for these

**Review by patriknw — COMMENTED** [review]
Looked at this again in more detail. Great work.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -14,30 +14,37 @@ object Dependencies {
...
+        // it is probably possible to remove the dependency if needed.
         case Some((2, n)) if n >= 13 => "0.9.0"
         case Some((2, n)) if n == 12 => "0.8.0"
         case _                       => "0.7.0"
       }
-    })
+    },
+    sslConfigVersion := {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, n)) if n >= 13 => "0.4.0"
```
Comment (jrudolph) [inline project/Dependencies.scala]: Why is that needed? Is 0.4.0 missing for other Scala version?

Comment (raboof) [reply project/Dependencies.scala]: I think this was just a 'conservative choice' because I wasn't sure 0.4.0 is binary compatible with 0.3.7, so I wasn't sure if updating to 0.4.0 for 2.11/2.12 would be an acceptable change.

**Comment by raboof** [comment]
continued as #26992
